gemini/genewise.py

```diff
@@ -0,0 +1,109 @@
+import sys
+import os
+from compiler import compile
+import operator
+import itertools as it
+from argparse import ArgumentParser
+from GeminiQuery import GeminiQuery
+
+
+def add_args(a=None):
+    if a is None:
+        a = ArgumentParser()
+    a.add_argument("--min-filters", type=int, default=1)
+    a.add_argument("--gt-filter", required=True, default=[], action='append')
+    a.add_argument("--filter")
+    a.add_argument("--columns", default="chrom,start,end,gene,impact,impact_severity,max_aaf_all")
+    a.add_argument("db")
+    return a
+
+
+def add_cols(cols, gt_filter):
+    assert isinstance(cols, list)
+    all_cols = ["gts", "gt_types", "gt_phases", "gt_depths",
+                "gt_ref_depths", "gt_alt_depths", "gt_quals",
+                "gt_phred_ll_homref", "gt_phred_ll_het",
+                "gt_phred_ll_homalt"]
+    return [x for x in all_cols if x in gt_filter and not x in cols]
+
+
+def gen_results(rows, gt_filters, min_filters, min_variants, columns):
+    # we track the index of the passed filter in passed_filters.
+    gene_passed_filters = {}
```

Comment:
I'll leave it as-is for now as it won't be a bottle-neck and avoids issues with starting at 0.